File: tests/fixtures/openai-realtime/openai-realtime-handle-error-server-event-fixed/02-adversarial-no-type-dispatch.ts
import WebSocket from 'ws';

// Adversarial: this handler doesn't branch on message.type at all (e.g. it
// forwards everything to a single downstream consumer), so there's no
// type-dispatch pattern to be missing an 'error' branch from.
function setup(forwardToClient: (raw: unknown) => void) {
  const url = 'wss://api.openai.com/v1/realtime?model=gpt-realtime';
  const callerSocket = new WebSocket(url, { headers: {} });

  callerSocket.on('message', (raw) => {
    forwardToClient(raw);
  });
}

File: tests/fixtures/openai-realtime/openai-realtime-handle-error-server-event-fixed/02_adversarial_no_type_dispatch.py
import websockets

# Adversarial: a handler with no message.type branching at all.
WS_URL = "wss://api.openai.com/v1/realtime?model=gpt-realtime"


async def stream():
    async with websockets.connect(WS_URL) as ws:
        async for raw in ws:
            append_to_transcript_log(raw)
